ocaml/typing/typecore.ml

```diff
@@ -3921,18 +3983,9 @@ let rec type_approx env sexp ty_expected =
   | Some (jexp, _attrs) -> type_approx_aux_jane_syntax ~loc env jexp ty_expected
   | None      -> match sexp.pexp_desc with
     Pexp_let (_, _, e) -> type_approx env e ty_expected
-  | Pexp_fun (lbl, def, pat, body) ->
-      let open Jane_syntax.N_ary_functions in
-      type_approx_function ~loc env
-        [ { pparam_desc = Pparam_val (lbl, def, pat);
-            pparam_loc = pat.ppat_loc
-          }
-        ]
-        None (Pfunction_body body) ty_expected
-  | Pexp_function cases ->
-      let open Jane_syntax.N_ary_functions in
-      type_approx_function ~loc
-        env [] None (Pfunction_cases (cases, sexp.pexp_loc, [])) ty_expected
+  | Pexp_fun _ | Pexp_function _ ->
+      Misc.fatal_error
+        "Unexpected [Pexp_fun]/[Pexp_function] outside of Jane Syntax construct"
```

Comment:
These cases are unrepresentable upstream, where `Pexp_fun` was removed and `Pexp_function` takes a list of params. These cases are only representable internally because we haven't changed the parsetree from 4.14. (And yes, we want to worry about and catch not-updated ppxes.)

Leaving this for you to resolve once you've checked your understanding against this.